.github/workflows/ci-pr.yml

```diff
@@ -0,0 +1,119 @@
+name: CI-PR
+on:
+  pull_request:
+    branches: [master]
+concurrency:
+  group: ci-${{ github.head_ref }}
+  cancel-in-progress: true
+jobs:
+  ci-pr:
+    name: CI-PR
+    needs: [test, msrv, lint]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Done
+        run: exit 0
+  test:
+    name: Tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "windows-latest"]
+        rust: ["stable"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - uses: Swatinem/rust-cache@v1
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Default features
+        run: cargo test --all-targets
+      - name: All features + Debug
+        run: cargo test --all-targets  --features "wrap_help yaml regex debug"
+      - name: No features
+        run: cargo test --all-targets --no-default-features --features "std cargo"
```

Comment:
Why are we running them all in one job instead of a matrix?

Reply:
It was an experiment to look into the trade off of sharing compilation between tests vs parallel running of tests.

Reply:
I've had great success using external tools to handle that i.e. [Toast](https://github.com/stepchowfun/toast) which uses container layer caching and has *drastically* sped up our CI at work. It's not all roses, beyond just adding an external tool, as you do have to twiddle a little bit layer dependencies to ensure the cache applies to all jobs you want to use it in and you need to watch out for when the cache *should* be invalid. But overall it's been a great addition. Not saying to implement it here, just something to keep the back of our minds for the future.